sdks/python/apache_beam/ml/rag/embeddings/base.py

```diff
@@ -0,0 +1,44 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from apache_beam.ml.transforms.base import EmbeddingTypeAdapter
+from apache_beam.ml.rag.types import Embedding
+
+
+def create_rag_adapter() -> EmbeddingTypeAdapter:
+  """Creates adapter for converting between Chunk and Embedding types.
+    
+    The adapter:
+    - Extracts text from Chunk.content.text for embedding
+    - Creates Embedding objects from model output
+    - Preserves Chunk.id and metadata in Embedding
+    - Sets sparse_embedding to None (dense embeddings only)
```

Comment:
I think that all we need to support sparse embeddings is to flip dense/sparse in `dense_embedding=embeddings, sparse_embedding=None`, right? At least in terms of this adapter? Should we just parameterize this?

Reply:
Another thought - if we enabled that use case, it would be hard to generate both dense and sparse embeddings in the same pipeline (because we're expecting a `Chunk`, but as soon as you do this once it generates an `Embedding`). It would be nice if this optionally accepted either a Chunk or an embedding.

I actually don't think we'd need to change any of the below code other than naming to make it obvious that this can happen.

Reply:
Yeah, I like this more - it also means we don't need to keep properties consistent across chunks/embeddings